doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,289 @@
+
+- For example
+    ```
+        /sonic_platform_common/sonic_y_cable/y_cable_vendor_mapping.py
+    ```
+
+Vendors can have several implementations/ways to use this concept
+- Vendor has a 1:1 mapping between model and Python file
+- Vendor has one Python file which supports all models
+- A combination of the above
+
+- Mapping could be such that its a dictionary in 2D
+
+    ```python
+    {
+         "<vendor_name_1>": {
+             "<model_name>": "<module>"
+         },
+         "<vendor_name_2>" : {
+             "<model_name>": "<module>"
+         }
+    }
+    ```
+
+- For example
+
+    ```python
+    {
+         "Vendor1": {
+             "model1": "vendor1/y_cable.py"
+         },
+
+         "Vendor2" : {
+             "model_a": "vendor2/y_cable_a.py",
+             "model_b": "vendor2/y_cable_b.py"
+         }
+    }
+    ```
+
+#### Rationale
+
+  - This is because both the container and host can easily access this path. And it would be easy to mount this on container (PMon container) as applicable.
+  - The sonic_y_cable package is installed in both host and pmon
+
+#### Implementation details
+- Once the xcvrd will start (daemon launched) or transceiver is connected (plugged in), in both cases xcvrd can determine the Vendor name and part number from the register specification and then it can appropriately read and parse the vendor name and part number from the spec using eeprom. Once it has this information it can then use this file to Load/import the appropriate module.
+
+### Directory Structure
+
+#### Background
+
+- We need to have an appropriate place where vendors can implement their modules and provide SONiC with an abstraction layer where PMon container and host can access the Y-Cable packages of all the vendors with ease
+
+  - Currently the assumption is Y-Cable API will be written in the Python language
+  - If a vendor has an existing library in a different language, they will need to either find a way to wrap/bind it in Python to align with the description below or (preferrably) provide a pure Python implementation
+  - Since sonic_y_cable (sonic-platform-common/sonic_y_cable) is already built as a package today (only for a single Y-Cable vendor with sonic-buildimage) vendors can also place their implementation in this directory itself.
+  - Also, a vendor can provide multiple files to support multiple cables/groups of cables. Importing the module from the mapping should import all the necessary implementation for the cable
+  - Vendors can put their common implementation across multiple modules in helper files which can be present in the vendor directory or inside another subdirectory. The second example below shows this approach.
+
+#### Proposed Solution
+
+- Vendors can place their implementations in this format
+
+    ```
+    sonic_platform_common/sonic_y_cable/<vendor>/<module>
```

Comment:
why can't we have vendor specific y_cable implementation in sonic-buildimage repo instead of platform-common repo (which shouldn't be having any changes/implementation which are platform specific)? For eg Sfpbase.py resides in platform-common whereas its realization/implementation reside in platform specific :- platform/broadcom/sonic-platform-modules-dell/z9264f/sonic_platform/sfp.py i.e in sonic-buildimage repo

Reply:
Hmm, platform is really a platform in a sense we have different SKU's, related configs etc, here these are just cables. Even though we could do something similar seems an overkill where it is easy to maintain everything in one place.